src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/caseupdate/services/CaseUpdaterTest.java

```diff
@@ -99,6 +100,128 @@ public void should_update_case_data_with_exception_record_if_no_warnings() {
+                new CaseUpdateRequest(
+                    false,
+                    transformationInput,
+                    null,
+                    caseDetails(originalCase)
+                )
+            );
+
+        // then
+        assertThat(result.caseUpdateDetails.caseData.address)
+            .isEqualToComparingFieldByField(exceptionRecordAddress);
+        assertCaseData(result.caseUpdateDetails.caseData, originalCase);
+        assertThat(result.warnings).isEmpty();
+        assertThat(result.caseUpdateDetails.eventId).isEqualTo(CaseUpdater.EVENT_ID);
+        assertScannedDocumentsFromExistingCase(
+            result.caseUpdateDetails.caseData.scannedDocuments,
+            caseScannedDocuments
+        );
+        assertScannedDocumentsFromExceptionRecord(
+            result.caseUpdateDetails.caseData.scannedDocuments,
+            exceptionRecordScannedDocuments
+        );
+    }
+
+    @Test
+    public void should_update_case_if_auto_process_is_false_and_update_details_warnings() {
+        // given
+        Address exceptionRecordAddress = address("-er");
+
+        List<Item<ScannedDocument>> caseScannedDocuments = scannedDocuments();
+        SampleCase originalCase = sampleCase(caseScannedDocuments);
+
+        List<InputScannedDoc> exceptionRecordScannedDocuments = inputScannedDocuments();
+        TransformationInput transformationInput = transformationInput(exceptionRecordScannedDocuments);
+
+        CaseUpdateDetails caseUpdateDetails = caseUpdateDetails(
+            exceptionRecordScannedDocuments,
+            emptyList()
+        );
+
+        given(caseUpdateDetailsValidator.getWarnings(caseUpdateDetails)).willReturn(asList("w1", "w2"));
+        given(addressExtractor.extractFrom(any())).willReturn(exceptionRecordAddress);
+
+        // when
+        SuccessfulUpdateResponse result =
+            caseUpdater.update(
+                new CaseUpdateRequest(
+                    false,
+                    transformationInput,
+                    caseUpdateDetails,
+                    caseDetails(originalCase)
+                )
+            );
+
+        // then
+        assertThat(result.caseUpdateDetails.caseData.address)
+            .isEqualToComparingFieldByField(exceptionRecordAddress);
+        assertCaseData(result.caseUpdateDetails.caseData, originalCase);
+        assertThat(result.warnings).containsExactlyInAnyOrder("w1", "w2");
+        assertThat(result.caseUpdateDetails.eventId).isEqualTo(CaseUpdater.EVENT_ID);
+        assertScannedDocumentsFromExistingCase(
+            result.caseUpdateDetails.caseData.scannedDocuments,
+            caseScannedDocuments
+        );
+        assertScannedDocumentsFromExceptionRecord(
+            result.caseUpdateDetails.caseData.scannedDocuments,
+            exceptionRecordScannedDocuments
+        );
+    }
+
+    @Test
+    public void should_throw_422_for_auto_process_is_true_when_update_details_validation_warnings() {
```

Comment:
nothing to do with 422 status coded in this layer.

Reply:
oh, yes